test_runner/regress/test_tenant_detach.py

```diff
@@ -167,3 +167,196 @@ def test_detach_while_attaching(
 
     with pg.cursor() as cur:
         cur.execute("SELECT COUNT(*) FROM foo")
+
+
+@pytest.mark.parametrize("remote_storage_kind", [RemoteStorageKind.NOOP, RemoteStorageKind.MOCK_S3])
+def test_ignored_tenant_reattach(
+    neon_env_builder: NeonEnvBuilder, remote_storage_kind: RemoteStorageKind
+):
+    neon_env_builder.enable_remote_storage(
+        remote_storage_kind=remote_storage_kind,
+        test_name="test_remote_storage_backup_and_restore",
+    )
+    env = neon_env_builder.init_start()
+    pageserver_http = env.pageserver.http_client()
+
+    ignored_tenant_id, _ = env.neon_cli.create_tenant()
+    tenant_dir = env.repo_dir / "tenants" / str(ignored_tenant_id)
+    tenants_before_ignore = [tenant["id"] for tenant in pageserver_http.tenant_list()]
+    tenants_before_ignore.sort()
+    timelines_before_ignore = [
+        timeline["timeline_id"]
+        for timeline in pageserver_http.timeline_list(tenant_id=ignored_tenant_id)
+    ]
+    files_before_ignore = [tenant_path for tenant_path in tenant_dir.glob("**/*")]
+
+    # ignore the tenant and veirfy it's not present in pageserver replies, with its files still on disk
+    pageserver_http.tenant_ignore(ignored_tenant_id)
```

Comment:
We should check that the tenant is no longer in the pageserver's memory after this returns.

tenant status and/or or timline list HTTP calls would be good candidates.

Reply:
Would the existing 

```python
tenants_after_ignore = [tenant["id"] for tenant in pageserver_http.tenant_list()]
    assert ignored_tenant_id not in tenants_after_ignore, "Ignored tenant should be missing"
    assert len(tenants_after_ignore) + 1 == len(
        tenants_before_ignore
    ), "Only ignored tenant should be missing"
```

be enough? I can sure call some method with the ignored tenant_id, but this way I'm also checking other tenants have not disappeared (there should be a default one in this test)

Reply:
Yeah, it's enough. Not sure whether it was always there and I missed it, or whether it was added since I added this comment.

Either way, it's good as it is.